core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -200,4 +200,28 @@ public String location() {
       return location;
     }
   }
+
+  class SetSnapshotRefs implements MetadataUpdate {
+    private final Map<String, SnapshotRef> updated;
+
+    public SetSnapshotRefs(Map<String, SnapshotRef> updated) {
+      this.updated = updated;
+    }
+
+    public Map<String, SnapshotRef> updated() {
+      return updated;
+    }
+  }
+
+  class RemoveSnapshotRefs implements MetadataUpdate {
```

Comment:
Like `RemoveSnapshot` and `SetSnapshotRef`, I would probably make this remove a single ref.